soc/arm/nxp_imx/mcimx6x_m4/Kconfig.defconfig.mcimx6x_m4

```diff
@@ -13,12 +13,9 @@ config SOC
 config FLOAT
 	default y
 
-if CLOCK_CONTROL
-
 config CLOCK_CONTROL_IMX_CCM
-	def_bool y
-
-endif # CLOCK_CONTROL
+	default y
+	depends on CLOCK_CONTROL
```

Comment:
Do you need the depends on here?

Reply:
Needs to appear on both definitions if the symbol should depend on `CLOCK_CONTROL` at least, because direct dependencies get ORed when you have multiple definitions.

In practice the direct dependencies will only matter if the symbol gets selected/implied though. Thought it might be nice to put it in to declare the dependency, even if that's unlikely.

Reply:
Ohh... looks like `CLOCK_CONTROL` isn't fixed in this case either, so it def. needs the `depends on`.

Ignore the select/imply thing, and think of it as `default y if CLOCK_CONTROL`. `depends on` gets propagated to all the properties, but it only applies to the definition it appears on. Without `depends on`, you'd get a plain `default y`, ignoring the value of `CLOCK_CONTROL`.

Reply:
An example might make it clearer:

Say you have this:

```
if A

if B

config FOO
	int
	default 1
	depends on C

endif

endif

config FOO
	default 2
	depends on D
```

That's the same as this:

```
config FOO
	int
	default 1
	depends on A && B && C

config FOO
	default 2
	depends on D
```

In turn, that's (almost, ignoring select/imply) the same as this:

```
config FOO
	int
	default 1 if A && B && C

config FOO
	default 2 if D
```

In turn, that's logically equivalent to this:

```
config FOO
	int
	default 1 if A && B && C
	default 2 if D
```

The crux is that `depends on`/`if` only applies to the definition location where it appears.

Multiple definition locations are pretty subtle and easy to mess up. Might be nice to do less of them in Zephyr at some point.